fil_fungible_token/src/token/types.rs

```diff
@@ -20,108 +20,148 @@ pub type Result<T, E> = std::result::Result<T, ActorError<E>>;
 /// A standard fungible token interface allowing for on-chain transactions that implements the
 /// FRC-XXX standard. This represents the external interface exposed to other on-chain actors
 ///
-/// Token authors should implement this trait and link the methods to standard dispatch numbers in
-/// their actor's `invoke` entrypoint. A standard helper (TODO) is provided to aid method dispatch.
-///
-/// TODO: make non-pseudo code
-/// ```
-/// //struct Token {}
-/// //impl FrcXXXToken for Token {
-/// //    ...
-/// //}
-///
-/// //fn invoke(params: u32) -> u32 {
-/// //    let token = Token {};
-///
-/// //    match sdk::message::method_number() {
-/// //        1 => constructor(),
-/// //        2 => token.name(),
-/// //        _ => abort!()
-/// //        // etc.
-/// //    }
-/// //}
-/// ```
+/// Token authors must implement this trait and link the methods to standard dispatch numbers (as
+/// defined by [FRC-0042](https://github.com/filecoin-project/FIPs/blob/master/FRCs/frc-0042.md)).
 pub trait FrcXXXToken<E> {
     /// Returns the name of the token
+    ///
+    /// Must not be empty
     fn name(&self) -> String;
 
     /// Returns the ticker symbol of the token
+    ///
+    /// Must not be empty. Should be a short uppercase string
     fn symbol(&self) -> String;
 
     /// Returns the total amount of the token in existence
+    ///
+    /// Must be non-negative. The total supply must equal the balances of all addresses. The total
+    /// supply must equal the sum of all minted tokens less the sum of all burnt tokens.ß
```

Comment:
```suggestion
    /// supply should equal the sum of all minted tokens less the sum of all burnt tokens.
```
This is relaxed in the FIP too. There's really very little we can constrain outside of the method signatures.